core/client/fs/src/main/java/alluxio/client/block/stream/NettyPacketWriter.java

```diff
@@ -378,16 +404,19 @@ private void updateException(Throwable e) {
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws IOException {
       Preconditions.checkState(acceptMessage(msg), "Incorrect response type %s.", msg);
-      RPCProtoMessage response = (RPCProtoMessage) msg;
+      RPCProtoMessage message = (RPCProtoMessage) msg;
       // Canceled is considered a valid status and handled in the writer. We avoid creating a
       // CanceledException as an optimization.
-      if (response.getMessage().asResponse().getStatus() != PStatus.CANCELED) {
-        CommonUtils.unwrapResponseFrom(response.getMessage().asResponse(), ctx.channel());
+      Protocol.Response response = message.getMessage().asResponse();
+      if (response.getStatus() != PStatus.CANCELED) {
+        CommonUtils.unwrapResponseFrom(response, ctx.channel());
       }
-
       try (LockResource lr = new LockResource(mLock)) {
-        mDone = true;
-        mDoneOrFailed.signal();
+        mFlushedOrDoneOrFailed.signal();
```

Comment:
Do we need to signal this if `!response.getMessage().equals("FLUSHED")`?

Reply:
`!response.getMessage().equals("FLUSHED")` equal to the situation that a Write Request is `DONE`, we should also signal `mFlushedOrDoneOrFailed`. If not signal, the flush() may not be able to wake up if one thread calls flush() and then another thread calls close().

Reply:
@LuQQiu I think the problem is we need to signal `mFlushedOrDoneOrFailed` (which should only need to be `mFlushedOrFailed`) in the `FlushListener`'s `operationComplete`.